llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp

```diff
@@ -1425,9 +1426,18 @@ SDValue SelectionDAGLegalize::ExpandExtractFromVectorThroughStack(SDValue Op) {
 
   if (!Ch.getNode()) {
     // Store the value to a temporary stack slot, then LOAD the returned part.
+    auto &MF = DAG.getMachineFunction();
+    auto &MFI = MF.getFrameInfo();
     StackPtr = DAG.CreateStackTemporary(VecVT);
-    Ch = DAG.getStore(DAG.getEntryNode(), dl, Vec, StackPtr,
-                      MachinePointerInfo());
+    int FI = cast<FrameIndexSDNode>(StackPtr.getNode())->getIndex();
+    MachinePointerInfo PtrInfo =
+        MachinePointerInfo::getFixedStack(DAG.getMachineFunction(), FI);
+
+    MachineMemOperand *StoreMMO =
+        MF.getMachineMemOperand(PtrInfo, MachineMemOperand::MOStore,
```

Comment:
@RKSimon We dont use the `MF.getMachineMemOperand` function call in `ExpandInsertToVectorThroughStack`. If you want I can create a helper function containing:
```
  int FI = cast<FrameIndexSDNode>(StackPtr.getNode())->getIndex();
  MachinePointerInfo PtrInfo =
        MachinePointerInfo::getFixedStack(DAG.getMachineFunction(), FI);
```
Is that okay?

Reply:
Alternatively could add a variant of DAG.CreateStackTemporary that returns the value and additionally the appropriate MMO. You would need to pass in the flags you want to use

Reply:
Can you check if the ExpandInsertToVectorThroughStack could use the getMachineMemOperand()? It might be using a variant of getStore that effectively does the same thing.

Reply:
I believe instertelement has similar issue: https://godbolt.org/z/9nov9adeK

Reply:
In which case please create a helper that returns the MachineMemOperand and we can create a followup patch for ExpandInsertToVectorThroughStack to use it to fix the insert case.

Reply:
Added a helper for the same!